transport/src/main/java/io/netty/channel/ChannelOption.java

```diff
@@ -85,10 +86,50 @@
             valueOf("DATAGRAM_CHANNEL_ACTIVE_ON_REGISTRATION");
 
     /**
-     * Creates a new {@link ChannelOption} with the specified {@code name}.
+     * Creates a new {@link ChannelOption} with the specified {@param name} or return the already existing
+     * {@link ChannelOption} for the given name.
      */
+    @SuppressWarnings("unchecked")
     public static <T> ChannelOption<T> valueOf(String name) {
-        return new ChannelOption<T>(name);
+        if (name == null) {
```

Comment:
Is there anyway to extract these new methods out and share this code with `AttributeKey`?  Perhaps a utility class with a factory member and an extra generic parameter would do the trick?